### PR TITLE
Closes #715 Prevent empty titles

### LINKED ISSUE #715: I shouldn't be able to rename a project to a blank string.
Currently, I'm able to save the project title even when the input is blank and I end up with this.

image

Let's at least force this basic validation. If someone tries to Save a blank project title, we just don't let them and the field stays editable until they type in some kind of name.

We should probably also determine if we want any restrictions on the project name
* Allow special characters?
* Max / min length?

I'm not too concerned, as long as we don't allow saving blanks.

cc @gideonthomas

### PR DESCRIPTION
**Pull request by errietta**

### CODE REVIEW

File: public/editor/scripts/editor/js/fc/project-rename.js [hunk 1]
```diff
@@ -109,6 +114,12 @@ define(function(require) {
     });
   }
 
+  function validate(context) {
```
Comment (sedge) [inline public/editor/scripts/editor/js/fc/project-rename.js]: I'd change this to `validateTitle`

**Comment by errietta** [comment]
@flukeout added a commit to disable btton when it's empty :smile: but the problem now is that if you click escape while on the input it empties :open_mouth: any idea what's the deal

**Comment by flukeout** [comment]
@errietta Will take a look!

**Comment by flukeout** [comment]
@errietta Yeah, you're right - but it's not a problem with your code! The production version of thimble behaves in the exact same way, I'll file it as a separate issue.

**Comment by flukeout** [comment]
@errietta In testing your patch, I found that I can still save a 0 length title if I use the enter key while editing.

**Comment by errietta** [comment]
Weird it didn't do thst to me. Ill take a look

On Fri, 2 Oct 2015 16:20 Luke Pacholski <notifications@github.com> wrote:

> @errietta <https://github.com/errietta> In testing your patch, I found
> that I can still save a 0 length title if I use the enter key while editing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/thimble.webmaker.org/pull/1159#issuecomment-145059616>
> .
>
-- 
Errietta Kostala
<errietta@errietta.me>

**Comment by errietta** [comment]
@flukeout i can't reproduce this, how do you manage it? I deleted the title and pressed enter but it didn't save?

**Comment by errietta** [comment]
@flukeout i fixed it now, R?

**Comment by flukeout** [comment]
@errietta Checking now!

**Comment by flukeout** [comment]
@errietta Tested this in Firefox and Chrome, and it works as expected! Great work! Let's leave it for @gideonthomas for a code review, but looks good from a UX perspective! :cake:

**Comment by flukeout** [comment]
Awesome work @errietta!

**Pull request merged**